src/kaltura-player.js

```diff
@@ -119,28 +119,34 @@ class KalturaPlayer extends FakeEventTarget {
     this.reset();
     this._localPlayer.loadingMedia = true;
     this._uiWrapper.setLoadingSpinnerState(true);
-    const providerResult = this._provider.getMediaConfig(mediaInfo);
-    providerResult
-      .then(
-        (providerMediaConfig: ProviderMediaConfigObject) => {
-          const mediaConfig = Utils.Object.copyDeep(providerMediaConfig);
-          if (mediaOptions) {
-            mediaConfig.sources = mediaConfig.sources || {};
-            mediaConfig.sources = Utils.Object.mergeDeep(mediaConfig.sources, mediaOptions);
+    return new Promise((resolve, reject) => {
```

Comment:
@dan-ziv Please change https://github.com/kaltura/kaltura-player-js/blob/master/src/common/playlist/playlist-manager.js#L286 to use the returning value instead of the config (sources and plugins)

Reply:
I didn't changed anything in the return value, just the way we returning it.

Reply:
what you returning now is containing the merge plugins and the redirect options so no need to take them from the config in the playlist manager